xbmc/guilib/Shader.cpp

```diff
@@ -102,9 +109,27 @@ bool CShader::InsertSource(const std::string& filename, const std::string& loc)
 
   m_source.insert(locPos, temp);
 
+  m_filenames.append(" " + filename);
+
   return true;
 }
 
+const std::string CShader::GetSourceWithLineNumbers()
```

Comment:
`std::string CShader::GetSourceWithLineNumbers() const` marks the function as const, meaning the function doesn't change the class state.
Your version just marks the return value const, which isn't of if returning by value.

Reply:
@Rechi thanks for the explanation 👍